tests/sklearn/test_sklearn_models.py

```diff
@@ -1120,6 +1137,200 @@ def check_exposition_structural_methods_decision_trees(model, x, y):
     )
 
 
+# pylint: disable=too-many-locals,too-many-statements
+@pytest.mark.parametrize("model_class, parameters", get_sklearn_tree_models_and_datasets())
+@pytest.mark.parametrize("use_rounding", [False, True])
+def test_load_fitted_sklearn_tree_models(
+    subtests,
+    model_class,
+    parameters,
+    use_rounding,
+    load_data,
+    is_weekly_option,
+    verbose=True,
+):
+    """Test `from_sklearn_model` functionnality of tree-based models."""
+
+    numpy.random.seed(0)
+    os.environ["TREES_USE_ROUNDING"] = str(int(use_rounding))
+
+    x, y = get_dataset(
+        model_class, parameters, min(N_BITS_REGULAR_BUILDS), load_data, is_weekly_option
+    )
+
+    if verbose:
+        print("Run check_load_pre_trained_sklearn_models")
+
+    assert issubclass(model_class, BaseTreeEstimatorMixin)
+    concrete_model = instantiate_model_generic(model_class, n_bits=min(N_BITS_REGULAR_BUILDS))
+    # Fit the model and retrieve both the Concrete ML and the scikit-learn models
+    with warnings.catch_warnings():
+        # Sometimes, we miss convergence, which is not a problem for our test
+        warnings.simplefilter("ignore", category=ConvergenceWarning)
+        concrete_model, sklearn_model = concrete_model.fit_benchmark(x, y)
+
+    # This step is needed in order to handle partial classes
+    model_class = get_model_class(model_class)
```

Comment:
if you use `is_model_class_in_a_list` in the following, you don't need to worry about partial classes (that's one of the benefit of using the util func)

Reply:
alright makes sense thanks

Reply:
On second thought I need the actual model class to be able to call `from_sklearn_model` 🤷🏼

Reply:
ah I see, indeed you have no choice here then !